app/permission/foreground/android.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+//+build android
+
+/*
+Package foreground implements permissions to run a foreground service.
+See https://developer.android.com/guide/components/foreground-services.
+
+The following entries will be added to AndroidManifest.xml:
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
+*/
+
+package foreground
+
+import (
+	"gioui.org/app"
+	"git.wow.st/gmp/jni"
```

Comment:
We already have a JNI implementation in package app. I think it should be re-used instead of adding a new dependency.

Reply:
ok, I didn't notice that - I copied gioui.org/x/notify's implementation - now I see this is available to package app so I will refactor here.

Reply:
this is a bit trickier to do than I thought at first because the package foreground wont see the unexported methods from app.

Reply:
I think perhaps it makes sense to move the startForeground() into the app package - in os_android.go

Reply:
You may have to refactor package app's JNI implementation into app/internal/jni.